### ISSUE TITLE
[Bug]maskrcnn.onnx  converted to IR

### ISSUE THREAD

**Issue by cheny985** [openvinotoolkit/openvino#13057]
##### System information (version)
- OpenVINO => 2022.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2019
- Problem classification: Model Conversion
- Framework: pytorch
- Model name:maskrcnn_resnet50_fpn_coco
##### Detailed description
The inference result of maskrcnn_resnet50_fpn_coco.pth is correct when it is converted to onnx, but after inferring to IR through the mo tool provided by openvion2022.1, the number of labels, boxes and scores of the model is correct, but the number of masks results is only 1, that is, the Batch_size of masks is 1, Only one mask information can be obtained.
onnx:![image](https://user-images.githubusercontent.com/67719986/190303428-b886b014-f0ed-4495-9271-d5955c8e410c.png)
IR:![image](https://user-images.githubusercontent.com/67719986/190303706-d0474e2c-cf47-44e1-9c18-ef8031a062e4.png)
<!--
    ```pth_to_onnx.py
    // import torchvision
import onnxruntime as ort
import numpy as np
import onnx
import torch
from network_files import MaskRCNN
from backbone import resnet50_fpn_backbone
backbone = resnet50_fpn_backbone(pretrain_path="resnet50.pth", trainable_layers=3)
model = MaskRCNN(backbone, num_classes=2)
model.load_state_dict(torch.load('./save_weight_old/save_weights8.25/model_169.pth',map_location='cpu')['model'])
model.eval()
data = torch.randn(100,3, 512, 512)
input_names = ["image"]
output_names = ['boxes', 'labels', 'scores', 'masks']
torch.onnx.export(model, data, "./mask_rcnn4.onnx",
                export_params=True,
                opset_version=11,
                input_names=input_names,
                output_names=output_names,
                dynamic_axes={
                    'image': [0, 2, 3],
                    "boxes": [0],
                    "labels": [0],
                    "scores": [0],
                    "masks": [0, 2, 3]
                }
                    )
    ```
    ```IR.C++
#include <openvino/openvino.hpp>
#include <opencv2/opencv.hpp>

using namespace cv;
using namespace std;

int main(int argc, char** argv) {
	
	ov::Core core;	
	Mat src = imread("C:\\Users\\chenyang\\Desktop\\1\\3.jpg");
	int im_h = src.rows;
	int im_w = src.cols;
	
	cv::RNG rng(12345);
	string model_path = "C:\\Users\\chenyang\\Desktop\\IR\\mask_rcnn9.xml";	
	ov::CompiledModel compiled_model = core.compile_model(model_path, "AUTO");
	ov::InferRequest infer_request = compiled_model.create_infer_request();
	ov::Tensor input_tensor = infer_request.get_input_tensor();
	ov::Shape tensor_shape = input_tensor.get_shape();
	std::cout << tensor_shape << std::endl;
	size_t channels = tensor_shape[1];
	size_t height = tensor_shape[2];
	size_t width = tensor_shape[3];

	Mat blob_image;
	resize(src, blob_image, Size(width, height));
	cvtColor(blob_image, blob_image, cv::COLOR_BGR2RGB);
	blob_image.convertTo(blob_image, CV_32F, 1.0 / 255);

	float* image_data = input_tensor.data<float>();
	std::cout << image_data << std::endl;
	for (size_t c = 0; c < channels; c++) {
		for (size_t h = 0; h < height; h++) {
			for (size_t w = 0; w < width; w++) {
				image_data[c * width * height + h * width + w] = blob_image.at<Vec3f>(h, w)[c];
			}
		}
	}
	infer_request.infer();

	auto labels = infer_request.get_tensor("labels");
	auto scores = infer_request.get_tensor("scores");
	auto boxes = infer_request.get_tensor("boxes");
	auto raw_masks = infer_request.get_tensor("masks");

	const float* out_labels = (float*)labels.data();
	const float* out_scores = (float*)scores.data();
	const float* out_boxes = (float*)boxes.data();
	const auto out_masks = (float*)raw_masks.data();
	std::cout << out_masks[0] << std::endl;

	ov::Shape label_out_shape = labels.get_shape();
	ov::Shape scores_out_shape = scores.get_shape();
	ov::Shape boxes_out_shape = boxes.get_shape();
	ov::Shape masks_out_shape = raw_masks.get_shape();
	
	std::cout << label_out_shape << std::endl;
	std::cout << scores_out_shape << std::endl;
	std::cout << boxes_out_shape << std::endl;
	std::cout << masks_out_shape << std::endl;

	const int max_count = label_out_shape[0];
	const int object_size = boxes_out_shape[1];
	int mask_h = masks_out_shape[2];
	int mask_w = masks_out_shape[3];

	std::cout << max_count << std::endl;
	std::cout << object_size << std::endl;
	std::cout << mask_h << std::endl;
	std::cout << mask_w << std::endl;
 ```

**Comment by Iffa-Meah** [comment]
Could you provide these:
1. Your model files for us to validate from our side
2. The commands and steps  that you used to convert model into IR

**Comment by cheny985** [comment]
1.model_169.pth is my training weight,mask_rcnn9.onnx is the onnx model.
2.The commands and steps ：mo --input_model C:/Users/chenyang/Desktop/1/mask_rcnn9.onnx --input_shape [1,3,512,512]
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "openvinotoolkit/openvino"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年9月15日(星期四) 中午12:43
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [openvinotoolkit/openvino] [Bug]maskrcnn.onnx  converted to IR (Issue #13057)





 
Could you provide these:
  
Your model files for us to validate from our side
 
The commands and steps  that you used to convert model into IR
  
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;


从QQ邮箱发来的超大附件

mask_rcnn9.onnx (167.78M, 2022年10月15日 13:52 到期)进入下载页面：http://mail.qq.com/cgi-bin/ftnExs_download?t=exs_ftn_download&k=7635313090bee796252dfbae4437564c57030655550750551f535207541a505a05561c525054554e0106080452530256545706516238640e53465a6f10540a0d0b1b5e5e0c4f645e&code=2510b7dc




maskrcnn_resnet50_fpn.pth (169.84M, 2022年10月15日 13:53 到期)进入下载页面：http://mail.qq.com/cgi-bin/ftnExs_download?t=exs_ftn_download&k=2b34643290bf5b90722caeac17300b4a5c01525650530d074804560a001d0d52000c490a54010f4854065c5305005804065101043129390804470f40525e573a1751175c54440c553a52145c1f404d0d6509&code=e4d2109e




model_169.pth (335.02M, 2022年10月15日 13:53 到期)进入下载页面：http://mail.qq.com/cgi-bin/ftnExs_download?t=exs_ftn_download&k=76656663fcd03f89237dacfd4565024d00070400055053521954075a004804010d5c4b01075c034f01505555075c0607010703526368300f5b01030f3c54065b1a15120b6358&code=4efcce0b

**Comment by avitial** [comment]
@cheny985 there was a similar issue reported and is currently under investigation, it seems some information about dynamic shapes is lost during conversion from ONNX to IR. I will share more details as they become available. 

Ref. 90554

**Comment by avitial** [comment]
@cheny985 this PR [https://github.com/openvinotoolkit/openvino/pull/10684](https://github.com/openvinotoolkit/openvino/pull/10684) might fix the issue you are observing, please check it out and give it a try on your side.

FYI I've performed a quick run with your provided onnx model and I can see the masks output layer is now dynamic. Let me know if you have any issues.

![image](https://user-images.githubusercontent.com/20826577/191860796-12c6368c-943b-4886-832f-bcc0043bede0.png)

**Comment by avitial** [comment]
PR has been merged to master branch, closing this issue. Feel free to reopen and ask additional questions related to this topic.
